### PR TITLE
Apply deadline RFC.

### PR DESCRIPTION
**Pull request by YoEight**
Changed: Apply gRPC deadlines RFC.
Changed: Default `NodePreference` to `Leader`.

### CODE REVIEW

File: src/grpc.rs [hunk 1]
```diff
@@ -122,6 +146,27 @@ where
...
+    S: Serializer,
+{
+    if let Some(value) = value.as_ref() {
+        serialize_duration(value, serializer)
+    } else {
+        serializer.serialize_none()
+    }
+}
+
+fn deserialize_optional_duration<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
```
Comment (oskardudycz) [inline src/grpc.rs:163]: **QUESTION:** Probably dumb question, but in which case we'll be deserialising this value? As I understood, we should be only sending it. Is it Rust convention to provide always both serialisation and deserialisation?

File: src/grpc.rs [hunk 2]
```diff
@@ -550,6 +601,33 @@ impl ClientSettings {
...
+                            let value = values.as_slice()[1];
+
+                            if let Ok(int) = value.parse::<i64>() {
+                                if int == -1 {
+                                    result.default_deadline = Some(Duration::from_millis(u64::MAX));
+                                    continue;
+                                }
+
+                                if int < -1 {
+                                    error!("Invalid defaultDeadline of {}. Please provide a positive integer, or -1 to disable", int);
```
Comment (oskardudycz) [inline src/grpc.rs:614]: **SUGGESTION:** I might be missing, but I don't see a negative test case for this error.

**Review by oskardudycz — APPROVED** [review]

File: src/grpc.rs (hunk 1 above)
Comment (YoEight) [reply src/grpc.rs:163]: We only do this when testing. There are fixtures/expectations we deserialize.

Comment (oskardudycz) [reply src/grpc.rs:163]: Hm 🤔 Could you point me to those tests?

File: src/types.rs [hunk 3]
```diff
@@ -1225,7 +1225,7 @@ pub enum NodePreference {
 
 impl Default for NodePreference {
     fn default() -> Self {
-        NodePreference::Random
+        NodePreference::Leader
```
Comment (oskardudycz) [inline src/types.rs:1228]: **FIX** Could you include this change in the PR description so we have this in the changelog? This is important, potentially breaking change.

File: src/grpc.rs (hunk 1 above)
Comment (YoEight) [reply src/grpc.rs:163]: https://github.com/EventStore/EventStoreDB-Client-Rust/blob/deadlines/src/grpc.rs#L43

**Review by arwinneil — APPROVED** [review]

**Pull request merged**